client/model.go

```diff
@@ -241,22 +243,56 @@ type TeamProjectAssignment struct {
 }
 
 type Policy struct {
-	Id                             string `json:"id"`
-	ProjectId                      string `json:"projectId"`
-	NumberOfEnvironments           int    `json:"numberOfEnvironments"`
-	NumberOfEnvironmentsPerProject int    `json:"numberOfEnvironmentsPerProject"`
-	RequiresApprovalDefault        bool   `json:"requiresApprovalDefault"`
-	IncludeCostEstimation          bool   `json:"includeCostEstimation"`
-	SkipApplyWhenPlanIsEmpty       bool   `json:"skipApplyWhenPlanIsEmpty"`
-	DisableDestroyEnvironments     bool   `json:"disableDestroyEnvironments"`
-	UpdatedBy                      string `json:"updatedBy"`
+	Id                         string `json:"id"`
+	ProjectId                  string `json:"projectId"`
+	NumberOfEnvironments       int    `json:"numberOfEnvironments"`
+	NumberOfEnvironmentsTotal  int    `json:"numberOfEnvironmentsTotal"`
+	RequiresApprovalDefault    bool   `json:"requiresApprovalDefault"`
+	IncludeCostEstimation      bool   `json:"includeCostEstimation"`
+	SkipApplyWhenPlanIsEmpty   bool   `json:"skipApplyWhenPlanIsEmpty"`
+	DisableDestroyEnvironments bool   `json:"disableDestroyEnvironments"`
+	SkipRedundantDepolyments   bool   `json:"skipRedundantDeployments"`
+	UpdatedBy                  string `json:"updatedBy"`
 }
 
 type PolicyUpdatePayload struct {
 	ProjectId                  string `json:"projectId"`
 	NumberOfEnvironments       int    `json:"numberOfEnvironments"`
+	NumberOfEnvironmentsTotal  int    `json:"numberOfEnvironmentsTotal"`
 	RequiresApprovalDefault    bool   `json:"requiresApprovalDefault"`
 	IncludeCostEstimation      bool   `json:"includeCostEstimation"`
 	SkipApplyWhenPlanIsEmpty   bool   `json:"skipApplyWhenPlanIsEmpty"`
 	DisableDestroyEnvironments bool   `json:"disableDestroyEnvironments"`
+	SkipRedundantDepolyments   bool   `json:"skipRedundantDeployments"`
+}
+
+func (p PolicyUpdatePayload) MarshalJSON() ([]byte, error) {
+	type serial struct {
```

Comment:
I tend to dislike this anti-pattern of arbitrary types. This is quite uncommon in golang.
Why can't we set`PolicyUpdatePayload` to simply support pointers instead?

Reply:
This would substantially complicate the code, because there would need to be `nil` checks everywhere these attributes are referenced. With 0 as an invalid value for these attributes that can be caught by the `ValidateDiagFunc`, I'm not sure what is gained by converting the `Policy` types to use pointers.

Reply:
This would substantially complicate the code, because there would need to be `nil` checks everywhere these attributes are referenced. With 0 as an invalid value for these attributes that can be caught by the `ValidateDiagFunc`, I'm not sure what is gained by converting the `Policy` types to use pointers.